pkg/pillar/cmd/volumemgr/handlework.go

```diff
@@ -134,155 +44,57 @@ type casIngestWorkResult struct {
-	delete(volumeWorkResultMap, key)
-}
-
-func lookupCasIngestWorkResult(ctx *volumemgrContext, key string) *casIngestWorkResult {
-	if res, ok := casIngestWorkResultMap[key]; ok {
-		return &res
-	}
-	return nil
-}
-
-func addCasIngestWorkResult(ctx *volumemgrContext, key string, res casIngestWorkResult) {
-	casIngestWorkResultMap[key] = res
-}
-
-func deleteCasIngestWorkResult(ctx *volumemgrContext, key string) {
-	delete(casIngestWorkResultMap, key)
-}
-
 // MaybeAddWorkCreate checks if the Key is in the map of pending work
 // and if not kicks of a worker and adds it
 // XXX defer if busy?
 func MaybeAddWorkCreate(ctx *volumemgrContext, status *types.VolumeStatus) {
-	log.Infof("MaybeAddWorkCreate(%s)", status.Key())
-	if lookupPendingCreate(ctx, status.Key()) {
-		log.Infof("MaybeAddWorkCreate(%s) found", status.Key())
-		return
-	}
 	d := volumeWorkDescription{
 		create: true,
 		status: *status,
 	}
-	w := worker.Work{Key: status.Key(), Description: d}
+	w := worker.Work{Kind: "create", Key: status.Key(), Description: d}
 	// XXX could check a return and not add...
 	ctx.worker.Submit(w)
-	// XXX success - add
-	addPendingCreate(ctx, status.Key())
-	log.Infof("MaybeAddWorkCreate(%s) done", status.Key())
 }
 
 // MaybeAddWorkLoad checks if the Key is in the map of pending work
 // and if not kicks of a worker and adds it
 // XXX defer if busy?
 func MaybeAddWorkLoad(ctx *volumemgrContext, status *types.ContentTreeStatus) {
-	key := status.Key()
-	log.Infof("MaybeAddWorkLoad(%s)", key)
-	if lookupPendingLoad(ctx, key) {
-		log.Infof("MaybeAddWorkLoad(%s) found", key)
-		return
-	}
 	d := casIngestWorkDescription{
 		status: *status,
 	}
-	w := worker.Work{Key: key, Description: d}
+	w := worker.Work{Kind: "load", Key: status.Key(), Description: d}
 	// XXX could check a return and not add...
 	ctx.worker.Submit(w)
-	// XXX success - add
-	addPendingLoad(ctx, key)
-	log.Infof("MaybeAddWorkLoad(%s) done", key)
 }
 
 // DeleteWorkCreate is called by user when work is done
 func DeleteWorkCreate(ctx *volumemgrContext, status *types.VolumeStatus) {
-	log.Infof("DeleteWorkCreate(%s)", status.Key())
-	if !lookupPendingCreate(ctx, status.Key()) {
-		log.Infof("DeleteWorkCreate(%s) NOT found", status.Key())
-		return
-	}
-	deletePendingCreate(ctx, status.Key())
-	log.Infof("DeleteWorkCreate(%s) done", status.Key())
+	ctx.worker.Cancel(status.Key())
 }
 
 // DeleteWorkLoad is called by user when work is done
 func DeleteWorkLoad(ctx *volumemgrContext, key string) {
-	log.Infof("DeleteWorkLoad(%s)", key)
-	if !lookupPendingLoad(ctx, key) {
-		log.Infof("DeleteWorkLoad(%s) NOT found", key)
-		return
-	}
-	deletePendingLoad(ctx, key)
-	log.Infof("DeleteWorkLoad(%s) done", key)
+	ctx.worker.Cancel(key)
 }
 
 // MaybeAddWorkDestroy checks if the Key is in the map of pending work
 // and if not kicks of a worker and adds it
 // XXX defer if busy?
 func MaybeAddWorkDestroy(ctx *volumemgrContext, status *types.VolumeStatus) {
```

Comment:
It is no longer a Maybe with the new behavior.

Reply:
Sure it is. If it submitted the work already, i.e. the key already exists, then it won't redo it.

Reply:
So how about renaming the three MaybeAddX functions to be called AddX?

Reply:
Done.